File: BE/avalon/app/schemas/student_comment.py
from pydantic import BaseModel
from datetime import datetime

class StudentCommentBase(BaseModel):
    student_id: int
    comment: str
    comment_user_id: int

class StudentCommentCreate(StudentCommentBase):
    pass

class StudentComment(StudentCommentBase):
    comment_id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: FE/AL-homepage/src/types/studentComment.ts
export interface StudentComment {
  comment_id: number
  student_id: number
  comment: string
  comment_user_id: number
  created_at: string // ISO 8601 date string
}

export interface CreateStudentCommentRequest {
  student_id: number
  comment: string
  comment_user_id: number
}
